schema.json

```diff
@@ -0,0 +1,39 @@
+{
+    "name": "xyz",
+    "version": "0.0.1",
+    "resources": {
+        "xyz:index:Random": {
+            "properties": {
+                "length": {
+                    "type": "integer"
+                },
+                "result": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "length",
+                "result"
+            ],
+            "inputProperties": {
```

Comment:
There's a lot of repetition here.  Can this be simplified down to something that is reasonable human-writable?  Or should we focus on authoring this in some other form?

Reply:
We could explore making `inputProperties` and `requiredInputs` optional and taking the relevant information from `properties` and `required` when the former are not present.

We do need to accommodate resources where the inputs and outputs are potentially disjoint or have different types/optionality, however--all of our current providers use this.

If we wanted to diverge further from JSON schema, we could use a more TF-style approach where all input properties must have a corresponding output property, which would allow us to mark certain properties as inputs. This is confusing with respect to optional outputs, however--that confusion has come up recently in our own offline conversation. What we have here sacrifices some ease of authorship for flexibility.

Reply:
I sort of wonder whether a YAML serialization would be materially more hand-editable - both for doc comments and for merging input properties into outputs.  Something to explore, but this is fine for now.